bin/collator/src/parachain/service.rs

```diff
@@ -950,32 +950,32 @@ where
         sc_client_api::StateBackend<BlakeTwo256>,
     Executor: sc_executor::NativeExecutionDispatch + 'static,
 {
-    let client_for_aura = client.clone();
-
-    let aura_verifier = move || {
-        let slot_duration =
-            cumulus_client_consensus_aura::slot_duration(&*client_for_aura).unwrap();
-
-        Box::new(sc_consensus_aura::build_verifier::<AuraPair, _, _, _>(
-            sc_consensus_aura::BuildVerifierParams {
-                client: client_for_aura.clone(),
-                create_inherent_data_providers: move |_, _| async move {
+    let verifier_client = client.clone();
+
+    let aura_verifier = cumulus_client_consensus_aura::build_verifier::<AuraPair, _, _, _>(
+        cumulus_client_consensus_aura::BuildVerifierParams {
+            client: verifier_client.clone(),
+            create_inherent_data_providers: move |parent_hash, _| {
+                let cidp_client = verifier_client.clone();
+                async move {
+                    let slot_duration = cumulus_client_consensus_aura::slot_duration_at(
+                        &*cidp_client,
+                        parent_hash,
+                    )?;
                     let timestamp = sp_timestamp::InherentDataProvider::from_system_time();
 
                     let slot =
-                    sp_consensus_aura::inherents::InherentDataProvider::from_timestamp_and_slot_duration(
-                        *timestamp,
-                        slot_duration,
-                    );
+                            sp_consensus_aura::inherents::InherentDataProvider::from_timestamp_and_slot_duration(
+                                *timestamp,
+                                slot_duration,
+                            );
 
                     Ok((slot, timestamp))
-                },
-                telemetry: telemetry_handle,
-                check_for_equivocation: sc_consensus_aura::CheckForEquivocation::Yes,
-                compatibility_mode: sc_consensus_aura::CompatibilityMode::None,
```

Comment:
I believe you added this change to reduce block production miss-rate for collators - now it's being reverted to the old approach. Won't it return the previous issue back then?

Reply:
I thought it enables new block verifier but it doesn't. It just runs a check which I want to disable because it makes node go OOM when upgrading to async backing

Reply:
The new verifier isn't public. Will do a PR to make it public. Until then this is all we have

Reply:
Ok, but I'm not sure I follow, sorry.

If we re-enable the old approach, won't the old issue come back?

I understand you're doing this because OOM issue with async backing, but what about non-async backing chains? Won't be see the return of the increased block miss rate?

Reply:
The block miss rate will be as is. This didn't change anything on Shiden

Reply:
More clear after our call. Approving.